### ISSUE TITLE
Slicing and indexing support negative steps

### ISSUE THREAD

**Issue by Seeker98** [pytorch/pytorch#99297]
### 🚀 The feature, motivation and pitch

At present pytorch is not compatible with the following code(pytorch 2.0), and will raise an error 'step must be greater than zero'
```
a=torch.arange(12).float()
print(a[::-1])
```
At the same time, this is always available in numpy. Is it possible this basic feature be add in pytorch in the near future?

### Alternatives

An alternative to this is using torch.flip(a, dims=[0]), but it would be great if [::-1] can be used in both 0d/1d and multidim tensors.

**Comment by ngimel** [comment]
No, this feature won't be added in the near future. It requires supporting negative strides and significant rework of internals.

**Comment by vadimkantorov** [comment]
Could the case of `a[::-1]` be wired to `a.flip(0)`? This would allow to be able to reverse an array in a polymorphic way without having to type `numpy.flip` or `torch.flip`, as numpy doesn't have an instance method `flip`
